### ISSUE TITLE
Refactoring

### ISSUE THREAD

**Issue by jammon** [1337-server/automatic-ripping-machine#33]
### tl;dr
I want to make the code easier to read and to understand. For this it should become more structured, some functionality should be moved into different functions, modules or classes and some names should be changed.

### Description
Currently the code is - out of obvious historical reasons - mostly in a linear, imperative style. E.g. `ripper/main.py` looks roughly like this:

```
import <whatever>
import ...
...

def some_function():
    ...

def some_other_function():
    ...

def main(some, parameters):
    # long
    # list
    # of 
    # detailed
    # instructions
    # and
    # if
        # constructs
    # with
        # many
        # branches
    # ...

if __name__ == "__main__":
    # do some setup
    # 10 loc about logging
    # get arguments
    # initialize some variable
    # initialize Job using that variable
    # more about logging
    if (something with cdrom_status):
        # 2 loc, finally exit
    if (something with logging):
        # exit
    
    # again logging
    
    # something with database
    # change 2 variables in job
    
    # again database in 5 loc
    
    # again logging in 8 loc

    # some database query about Jobs

    # do something with the result in 10 loc

    # again logging

    try:
        main(logfile, job)
    except Exception as e:
        # logging
        # notifying
        # set job variables
    else:
        # set job variable
    finally:
        # 6 loc calculations
        # something with database
```

For somebody new to the project (like me) this is hard to grasp and to find my way through it. It is not easy to find the spot in the code, where the concept I'm looking for is dealt with. Some variables or functions have names that don't speak to me.

I want to refactor the code, so that it is more structured and divided in more obvious sections. Anything that modifies e.g. a job should be in a method of Job. Probably it would be wise to subclass Job for music, dvd, blueray etc. to get rid of long if-elif-statements.

`ripper/main.py` should then maybe look like this:

```
import <whatever>
import ...
...

def main(some, parameters):
    do_some_setup()
    initialize_logging()
    manage_previous_jobs_in_database()
    job = job_factory()
    job.do_the_ripping()
    maybe_write_something_to_the_database()
    log_exit_status()

def do_some_setup():
    ...

def initialize_logging():
    ...

if __name__ == "__main__":
    main()
```

Maybe it could also be a good idea to add the name of the function to the logging if it's a kind of error message.

Now this was mostly about `ripper/main.py`, but the other parts of the code should be treated in the same way.

### Your thoughts?
Since this involves a lot of changes to existing code, I don't want to do this against the current contributors. I tried to use this project some two or three years ago, but failed. I got errors that I couldn't solve, because I hadn't the time and patience to work my way through all this code. Now I started again because I 'inherited' a lot of music CDs that I want to rip. I'm willing to do this refactoring, but only if you will go with me in that process. Doing it for me alone seems not worth while.

To be clear: I'm not asking for your active participation in this process. I just want to know if you would consider this as an improvement and if you'd be willing to accept these pull requests. You can see my first attempts at https://github.com/jammon/automatic-ripping-machine/tree/readability.

Kind regards, Johannes

**Comment by jammon** [comment]
Thanks for your interest and your help!

If that is ok for you I will make pull request in short intervalls to keep this branch in sync with your development of the functionality.

**Comment by 1337-server** [comment]
That sounds great.

**Comment by jammon** [comment]
Ah, what a pity! I'm not at home for some days and I cannot get my raspberry pi running in this environment here, so I cannot test my code. So pull requests will have to wait until next week. :-(

**Comment by 1337-server** [comment]
There is no rush don't worry.

**Comment by jammon** [comment]
Ok, I managed to access the raspi, so I can go on :-)
